container_workflow_tool/dockerfile.py

```diff
@@ -0,0 +1,86 @@
+# MIT License
+#
+# Copyright (c) 2020 SCL team at Red Hat
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import re
+
+from container_workflow_tool.utility import setup_logger, RebuilderError
+
+
+class DockerfileHandler:
+    """Class for handling with Dockerfile files."""
+
+    def __init__(self, base_image, logger=None):
+        self.base_image = base_image
+        self.logger = logger if logger else setup_logger("dockerfile")
+
+    def get_from_df(self, dockerfile_path):
+        with open(dockerfile_path) as f:
+            fdata = f.read()
+        return self.get_from(fdata)
+
+    def get_from(self, fdata: str):
+        """Gets FROM field from a Dockerfile
+
+        Args:
+            fdata (str): String containing the Dockerfile
+
+        Returns:
+            str: FROM string
+        """
+        image_base = re.search('FROM (.*)\n', fdata)
+        if image_base.group(1) == "":
```

Comment:
This will fail if `re.search` returns None. What about:
```python
if image_base and image_base.group(1) != "":
    return image_base.group(1)
else:
    raise RebuilderError("FROM field is missing or the image name is empty")
```
Or, if an existing `FROM` without any string after a space is unlikely to happen, you can do:
```python
try:
    return image_base.group(1)
except AttributeError:
    raise RebuilderError("FROM field is missing")
```

Reply:
https://github.com/sclorg/container-workflow-tool/pull/73/commits/43ecfda3be4dbbc46df7f7520fca44013e422d3d